### ISSUE TITLE
Fallback mode

### ISSUE THREAD

**Issue by nullchinchilla** [niwl/geph#3]
It would be nice to have a _fallback mode_ (marketing name TBD) where all traffic is routed through domain fronting using my warpfront protocol. This can greatly increase reliability from "Lantern-like" to "effectively invulnerable to blocking", setting us apart from the competition. "When nothing else works, Geph will work" would be a great reputation to have in China etc, and can capture the market of people who mainly interact online using blocked services (anonymous political activists, expats, etc). 

In addition, the very high efficiency of warpfront both in terms of overhead and number of requests is possibly of some academic interest, and having a live implementation with real users is good. Existing domain-fronting protocols incur horrible delays (meek), are special-cased for HTTP (Lantern's current domain-fronting protocol), or spam a massive number of small expensive requests with large overhead (Lantern's proposed protocol).

Code needs to be added all over the place:
- ~~A flag `--cdn-front <url>` needs to be added to `geph entry`, specifying a domain-fronting-capable CDN front (like `https://blahblahblah.cloudfront.net`). The entry node would not open the usual ClutterShirt socket, but instead print a message like `You must forward <url> to http://<my-ip>:2379 before this fronted entry goes online`, and test `<url>` until it works, at which point it uploads `<url>` to the chosen exit node.~~
- `geph exit` needs to expose an interface to obtain a front/host pair. This would be distributed to all clients, since they cannot be blocked by knowing the URL.
- `geph client` needs a way of falling back (obviously) to domain fronting when all normal nodes fail. A `FallbackConn` state needs to be added to the machine. `netinfo` on RPC would display a different protocol indicating that the node used is a fallback node, so the UI can display this information.

Also, fallback mode will cost us significantly more than normal mode per GB; this can possibly be reflected by, say, charging two times more on bandwidth used in fallback mode. The user, of course, would be able to set settings like "never use fallback mode even if standard mode fails".

Fallback mode shouldn't be needed the vast majority of the time though; the mode would ideally activate only during events like a new attack from the GFW that blocks a significant proportion of our normal entry nodes before we can figure out defenses, or behind corporate firewalls that only allow HTTP and HTTPS. We would also want it to activate if the only entry nodes available are detected to have extremely poor speed.

Finally, fallback nodes can be placed in locations much more flexibly than normal nodes; we can use sketchy VPS providers in bad locations without worrying too much, as the CDN usually has routes good enough that pipes can get filled.

**Comment by nullchinchilla** [comment]
This is quite important really, as it will really boost resilience to blocking. It's also not hard to implement.

**Comment by nullchinchilla** [comment]
Done.
